File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        /*
        12345 (5)
        212 324 25 (8)
        33 11 22 44 55 (10)
        
        정답맞춘개수배열
        정답맞춘개수세기 :
            answer 인덱스 :
                
        */
        int[] c1 = {1,2,3,4,5};
        int[] c2 = {2,1,2,3,2,4,2,5};
        int[] c3 = {3,3,1,1,2,2,4,4,5,5};
        int[] check = new int[3];
        for (int i = 0; i<answers.length; i++){
            if (answers[i] == c1[i%5]){
                check[0] += 1;
            }
            if (answers[i] == c2[i%8]){
                check[1] += 1;
            }
            if (answers[i] == c3[i%10]){
                check[2] += 1;
            }
        }
        int maxVal = check[0];
        for (int i = 1; i<3; i++){
            maxVal = Math.max(check[i],maxVal);
        }
        ArrayList<Integer> ar = new ArrayList<>();
        for (int i = 0; i<3; i++){
            if (check[i] == maxVal){
                ar.add(i+1);
            }
        }
        int i = 0;
        int[] ans = new int[ar.size()];
        for (int x: ar){
            ans[i++] = x;
        }
        return ans;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    p1 = [1,2,3,4,5] # 5
    p2 = [2,1,2,3,2,4,2,5] # 8
    p3 = [3,3,1,1,2,2,4,4,5,5] # 10
    ans1,ans2,ans3 = 0,0,0
    
    for i,v in enumerate(answers):
        if v == p1[i%5]: ans1 += 1
        if v == p2[i%8]: ans2 += 1
        if v == p3[i%10]: ans3 += 1
    maxx = max(ans1,ans2,ans3)
    answer = []
    if ans1 == maxx: answer.append(1)
    if ans2 == maxx: answer.append(2)
    if ans3 == maxx: answer.append(3)
    return answer